planning/scheduler.go

```diff
@@ -0,0 +1,111 @@
+package planning
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ipfs-shipyard/w3rc/contentrouting"
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+)
+
+// ErrNoTransport is an error option a transport plan my emit when no transports are currently possible
+var ErrNoTransport = fmt.Errorf("no routes available")
+
+// Scheduler provides an interface for deciding which potential transports to begin when.
+type Scheduler interface {
+	// plan the fetch of a root+selector of data given a set of learned routing records.
+	Schedule(ctx context.Context, root cid.Cid, selector ipld.Node, potentialTransports <-chan contentrouting.RoutingRecord) <-chan TransportPlan
+	// Indicate that a transfer in the schedule has started
+	Begin(r *TransportRequest)
+	// Indicate that a transfer in the schedule has completed
+	Reconcile(r *TransportRequest, success bool)
```

Comment:
the common case will have lots of different requests. it's unclear that trying to combine all of them in the board makes sense.

you would like to know if all of them fail because e.g. you're offline. in the common case of some providers failing, there's not a great aggregate error you'd collect here. I also don't think it's something the board / scheduler needs to have the details/complexity of reasoning about.